manager/controllers/app/policy_compiler_interface.go

```diff
@@ -6,75 +6,58 @@ package app
 import (
 	"log"
 
-	"emperror.dev/errors"
 	app "fybrik.io/fybrik/manager/apis/app/v1alpha1"
 	"fybrik.io/fybrik/manager/controllers/utils"
 	connectors "fybrik.io/fybrik/pkg/connectors/clients"
-	pb "fybrik.io/fybrik/pkg/connectors/protobuf"
+	openapiclientmodels "fybrik.io/fybrik/pkg/taxonomy/model/base"
 	"fybrik.io/fybrik/pkg/vault"
 )
 
-// ConstructApplicationContext constructs ApplicationContext structure to send to Policy Compiler
-func ConstructApplicationContext(datasetID string, input *app.FybrikApplication, operation *pb.AccessOperation) *pb.ApplicationContext {
+func ConstructOpenAPIReq(datasetID string, input *app.FybrikApplication, operation *openapiclientmodels.PolicyManagerRequestAction) (*openapiclientmodels.PolicyManagerRequest, string, error) {
+	req := openapiclientmodels.PolicyManagerRequest{}
+	action := openapiclientmodels.PolicyManagerRequestAction{}
+	resource := openapiclientmodels.Resource{}
+
+	resource.SetName(datasetID)
+	req.SetResource(resource)
+
+	destination := operation.GetDestination()
+	action.SetDestination(destination)
+	operationType := operation.GetActionType()
+	if operationType == openapiclientmodels.READ {
+		action.SetActionType(openapiclientmodels.READ)
+	}
+	if operationType == openapiclientmodels.WRITE {
+		action.SetActionType(openapiclientmodels.WRITE)
+	}
+	action.SetProcessingLocation(operation.GetDestination())
+	req.SetAction(action)
+
+	reqContext := make(map[string]interface{})
+	for k, v := range input.Spec.AppInfo {
+		reqContext[k] = v
+	}
+	req.SetContext(reqContext)
+
 	var credentialPath string
 	if input.Spec.SecretRef != "" {
 		credentialPath = utils.GetVaultAddress() + vault.PathForReadingKubeSecret(input.Namespace, input.Spec.SecretRef)
 	}
```

Comment:
credentialPath should be constructed outside this function, it should not be returned by this function as well.

Reply:
fixed. Thanks.